### PR TITLE
Add qpOASES recipe

### PR DESCRIPTION
**Pull request by traversaro**
This recipe adds the qpOASES QP solver, quite used in scientific literature (see https://scholar.google.com/scholar?cites=12703661245711008199&as_sdt=2005&sciodt=0,5&hl=en) including in robotics. At the moment the recipe just contains the C++ package.

The package contain quite a big patch that contains the bugfixes contained in the master branch. This was necessary as upstream did not made a release recently, but as soon as a new release is done (see https://github.com/coin-or/qpOASES/issues/113) the patch can be dropped and the new release used as it is.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by traversaro** [comment]
CI is happy and the PR is ready for review. @conda-forge/help-c-cpp

File: recipes/qpoases/build.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+
```
Comment (chrisburr) [inline recipes/qpoases/build.sh:2]: As you're patching the version I'd suggest validating it if you can:
```suggestion
# Ensure the version is correct in the sources
grep 'PACKAGE_VERSION "'$PKG_VERSION'"' CMakeLists.txt

```

Comment (traversaro) [reply recipes/qpoases/build.sh:2]: Thanks, this would be useful also to check that in future the version is correctly updated.

**Comment by traversaro** [comment]
This recipe is quite used in robotics applications, fyi @Tobias-Fischer @wolfv .

**Comment by traversaro** [comment]
Thanks @wolfv!

**Pull request merged**